index.js

```diff
@@ -835,6 +871,68 @@ class Batch {
   }
 }
 
+class Watcher extends EventEmitter {
+  constructor (bee, range) {
+    super()
+
+    this.bee = bee
+    this.core = bee.core
+
+    this.running = false
+    this.destroyed = false
+
+    this.range = range
+    this.latestDiff = this.bee.version
+    this.stream = null
+
+    this._onappend = this._onappend.bind(this)
+    this._onappend = debounceify(this._onappend)
+
+    // this.on('error', noop)
+  }
+
+  async _onappend () {
+    this.running = true
+
+    try {
+      await this._run()
+    } catch (err) {
+      this.emit('error', err)
```

Comment:
i would just emit warning here instead of error (solves your question above).

Reply:
Wouldn't be better to just noop the `error` event? As it's a more common event, also notice that the error should only happen if user misuses the watcher

For example, no watch related, this throws `SESSION_CLOSED`:
```js
const put1 = db.put('asd')
await db.close()
await put1
```
Because let's say user is misusing `put()`

Example of watch throwing an error:
https://github.com/holepunchto/hyperbee/blob/bb99190279b2263bef26848a3b318f074b543f8a/test/watch.js#L216